gensim/models/nmf.py

```diff
@@ -0,0 +1,615 @@
+"""Online Non-Negative Matrix Factorization."""
+
+import itertools
+
+import logging
+import numpy as np
+import scipy.sparse
+from scipy.stats import halfnorm
+
+from gensim import interfaces
+from gensim import matutils
+from gensim import utils
+from gensim.interfaces import TransformedCorpus
+from gensim.models import basemodel, CoherenceModel
+from gensim.models.nmf_pgd import solve_h, solve_r
+
+logger = logging.getLogger(__name__)
+
+
+class Nmf(interfaces.TransformationABC, basemodel.BaseTopicModel):
+    """Online Non-Negative Matrix Factorization.
+
+    `Renbo Zhao et al :"Online Nonnegative Matrix Factorization with Outliers" <https://arxiv.org/abs/1604.02634>`_
+
+    """
+
+    def __init__(
+        self,
+        corpus=None,
+        num_topics=100,
+        id2word=None,
+        chunksize=2000,
+        passes=1,
+        lambda_=1.0,
+        kappa=1.0,
+        minimum_probability=0.01,
+        use_r=False,
+        w_max_iter=200,
+        w_stop_condition=1e-4,
+        h_r_max_iter=50,
+        h_r_stop_condition=1e-3,
+        eval_every=10,
+        v_max=None,
+        normalize=True,
+        sparse_coef=3,
+        random_state=None,
```

Comment:
missing test for `random_state` (train two models with same seed & check that they have identical matrices)

Reply:
Added test